code/__HELPERS/text.dm

```diff
@@ -393,17 +393,21 @@ proc/checkhtml(var/t)
 	if(parts.len==2)
 		. += ".[parts[2]]"
 
-var/global/list/watt_suffixes = list("W", "KW", "MW", "GW", "TW", "PW", "EW", "ZW", "YW")
+var/list/watt_suffixes = list("W", "KW", "MW", "GW", "TW", "PW", "EW", "ZW", "YW")
 /proc/format_watts(var/number)
-	if(number<0)
-		return "-[format_watts(number)]"
-	if(number==0)
+	if (number<0)
+		return "-[format_watts(abs(number))]"
+	if (number==0)
 		return "0 W"
 
+	var/max_watt_suffix = watt_suffixes.len
```

Comment:
wait aren't BYOND arrays zero-indexed

Reply:
No, they aren't

Reply:
It's still able to exceed the bounds of the list by 1
```dm
i = 1
i += watt_suffixes.len
```
congrats it's now out of bounds, and that's exactly what the while loop will cause

Reply:
should be fine then
it can't exceed the bounds due to the if check in while